### PR TITLE
BlogServiceRemoteREST: Fixing settings parsing crash

### LINKED ISSUE #4187: Crash: BlogServiceRemoteREST.remoteBlogSettingFromJSONDictionary
```
2015-08-21 18:54:46.143 WordPress[98641:11654357] -[__NSCFBoolean isEqualToString:]: unrecognized selector sent to instance 0x6a72f60
2015-08-21 18:54:46.147 WordPress[98641:11654357] *** Terminating app due to uncaught exception 'NSInvalidArgumentException', reason: '-[__NSCFBoolean isEqualToString:]: unrecognized selector sent to instance 0x6a72f60'
*** First throw call stack:
(
    0   CoreFoundation                      0x067f4746 __exceptionPreprocess + 182
    1   libobjc.A.dylib                     0x0647da97 objc_exception_throw + 44
    2   CoreFoundation                      0x067fc705 -[NSObject(NSObject) doesNotRecognizeSelector:] + 277
    3   CoreFoundation                      0x06743287 ___forwarding___ + 1047
    4   CoreFoundation                      0x06742e4e _CF_forwarding_prep_0 + 14
    5   WordPress                           0x0024a159 -[BlogServiceRemoteREST remoteBlogSettingFromJSONDictionary:] + 713
    6   WordPress                           0x00248350 __61-[BlogServiceRemoteREST syncSettingsForBlog:success:failure:]_block_invoke + 256
    7   WordPress                           0x005ac59b __64-[AFHTTPRequestOperation setCompletionBlockWithSuccess:failure:]_block_invoke46 + 43
    8   libdispatch.dylib                   0x072995ea _dispatch_call_block_and_release + 15
    9   libdispatch.dylib                   0x072bbbef _dispatch_client_callout + 14
    10  libdispatch.dylib                   0x072a16bb _dispatch_main_queue_callback_4CF + 993
    11  CoreFoundation                      0x0674d8ee __CFRUNLOOP_IS_SERVICING_THE_MAIN_DISPATCH_QUEUE__ + 14
    12  CoreFoundation                      0x0670b5f0 __CFRunLoopRun + 2256
    13  CoreFoundation                      0x0670aa5b CFRunLoopRunSpecific + 443
    14  CoreFoundation                      0x0670a88b CFRunLoopRunInMode + 123
    15  GraphicsServices                    0x07f802c9 GSEventRunModal + 192
    16  GraphicsServices                    0x07f80106 GSEventRun + 104
    17  UIKit                               0x043090b6 UIApplicationMain + 1526
    18  WordPress                           0x00104d39 main + 73
    19  libdyld.dylib                       0x072e6ac9 start + 1
)
libc++abi.dylib: terminating with uncaught exception of type NSException
```

### PR DESCRIPTION
**Pull request by jleandroperez**
#### Steps:
1. Launch WPiOS
2. Tap over My Sites
3. Open any of your blogs
4. Tap over the `Settings` row

As a result, the app _should not_ crash.

Fixes #4187
Needs Review: @diegoreymendez  || @sendhil

### CODE REVIEW

File: WordPress/Classes/Networking/BlogServiceRemoteREST.m [hunk 1]
```diff
@@ -210,20 +210,22 @@ - (NSDictionary *)mapPostFormatsFromResponse:(id)response
...
-    } else {
-        remoteSettings.defaultCategory = @(PostCategoryUncategorized);
-    }
-    if ([json[@"settings"][@"default_post_format"] isEqualToString:@"0"]) {
+    remoteSettings.defaultCategory = [rawSettings numberForKey:@"default_category"] ?: @(PostCategoryUncategorized);
+
+    // Note:
+    // YES, the backend might send '0' as a number, OR a string value.
+    //
+    if ([[rawSettings numberForKey:@"default_post_format"] isEqualToNumber:@(0)] ||
```
Comment (diegoreymendez) [inline WordPress/Classes/Networking/BlogServiceRemoteREST.m]: You can use a constant for the key (even if it ends up being declared locally its better than the raw string).

File: WordPress/Classes/Networking/BlogServiceRemoteREST.m [hunk 2]
```diff
@@ -210,20 +210,22 @@ - (NSDictionary *)mapPostFormatsFromResponse:(id)response
...
-        remoteSettings.defaultCategory = [json numberForKeyPath:@"settings.default_category"];
-    } else {
-        remoteSettings.defaultCategory = @(PostCategoryUncategorized);
-    }
-    if ([json[@"settings"][@"default_post_format"] isEqualToString:@"0"]) {
+    remoteSettings.defaultCategory = [rawSettings numberForKey:@"default_category"] ?: @(PostCategoryUncategorized);
+
+    // Note:
+    // YES, the backend might send '0' as a number, OR a string value.
+    //
```
Comment (diegoreymendez) [inline WordPress/Classes/Networking/BlogServiceRemoteREST.m]: I'd consider adding a link to the ticket you're solving in the comments - for extra reinforcement of why this is being done. (https://github.com/wordpress-mobile/WordPress-iOS/issues/4187)

**Comment by jleandroperez** [comment]
@diegoreymendez comments addressed (thanks for the feedback!!).

Ready for a second round sir!

**Comment by diegoreymendez** [comment]
:shipit:

**Comment by jleandroperez** [comment]
Thank you sir!

**Pull request merged**